src/azure-cli/azure/cli/command_modules/util/custom.py

```diff
@@ -188,28 +188,33 @@ def _upgrade_on_windows():
     This also gives the user a chance to manually install the MSI in case of msiexec.exe failure.
     """
     import platform
+    import subprocess
+    import sys
+    import tempfile
+
+    from pathlib import Path
+    from azure.cli.core.util import rmtree_with_retry
 
     if platform.architecture()[0] == '32bit':
         msi_url = 'https://aka.ms/installazurecliwindows'
     else:
         msi_url = 'https://aka.ms/installazurecliwindowsx64'
     logger.warning("Updating Azure CLI with MSI from %s", msi_url)
-    tmp_dir, msi_path = _download_from_url(msi_url)
 
-    logger.warning("Installing MSI")
-    import subprocess
-    exit_code = subprocess.call(['msiexec.exe', '/i', msi_path])
+    # Save MSI to ~\AppData\Local\Temp\azure-cli-msi, clean up the folder first
+    msi_dir = Path(tempfile.gettempdir()) / 'azure-cli-msi'
```

Comment:
Azure CLI doesn't really use `pathlib`. Consider using `os.path` to be consistent with the rest of the code.

Reply:
`pathlib` is the recommended way to deal with path operations now.
> If you’ve never used this module before or just aren’t sure which class is right for your task, [Path](https://docs.python.org/3/library/pathlib.html#pathlib.Path) is most likely what you need.
> For low-level path manipulation on strings, you can also use the [os.path](https://docs.python.org/3/library/os.path.html#module-os.path) module.    --- https://docs.python.org/3/library/pathlib.html